test/password.cpp

```diff
@@ -0,0 +1,22 @@
+#include "catch.hpp"
+
+#include <password.h>
+
+using namespace newsbeuter;
+
+TEST_CASE("read_password()", "[password]") {
+    REQUIRE(read_password("data/empty.xml") == "");
+    REQUIRE(read_password("data/not-existing.txt") == "");
```

Comment:
1. There's a slight chance such a file will be created, especially under `data` directory. I suggest:

    a) taking it out of `data`, because the chances of text file existing inside `tests` directory is smaller;

    b) giving it a long, specific name like "a-passwordfile-that-is-guaranteed-to-not-exist.txt".

2. I'd add another test for non-existent file, one that checks that the function doesn't throw an exception in this case. Catch has a macro for that called `REQUIRE_NOTHROW`.